controllers/logging/secret_controller.go

```diff
@@ -0,0 +1,89 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/log"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	loggingv1 "github.com/openshift/elasticsearch-operator/apis/logging/v1"
+	"github.com/openshift/elasticsearch-operator/internal/k8shandler"
+)
+
+// SecretReconciler reconciles a Secret object
+type SecretReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+func (r *SecretReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+
+	cluster := &loggingv1.Elasticsearch{}
+	esName := types.NamespacedName{
+		Namespace: req.Namespace,
+		Name:      req.Name,
+	}
+
+	err := r.Get(context.TODO(), esName, cluster)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if req.Namespace != cluster.Namespace || req.Name != cluster.Name {
+		log.Info("ES secret namespace name mismatch")
+		return ctrl.Result{}, nil
+	}
+
+	err = k8shandler.SecretReconcile(cluster, r.Client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func esSecretUpdatePredicate(r client.Client) predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			esList := &loggingv1.ElasticsearchList{}
+			err := r.List(context.TODO(), esList)
+			if err != nil {
+				return false
+			}
+			for _, es := range esList.Items {
+				if e.MetaOld.GetNamespace() == es.Namespace && e.MetaOld.GetName() == es.Name {
+					return true
+				}
+			}
```

Comment:
i think we can actually improve the speed on this if we instead try to get a `loggingv1.Elasticsearch{}` object specifying the name and namespace as `e.MetaNew.GetName()` and `e.MetaNew.GetNamespace()`

Then we can check to see if we were returned an object, if so -- we return true, else we return false